src/Native/CpuMathNative/Sse.cpp

```diff
@@ -60,17 +60,171 @@ const unsigned int TrailingAlignmentMask[16] =
 // Multiply matrix times vector into vector.
 EXPORT_API(void) MatMul(_In_ const float * pmat, _In_ const float * psrc, _Inout_ float * pdst, int crow, int ccol)
 {
+    if (ccol < 4)
+    {
+        for (int i = 0 ; i < crow; i++)
+        {
+            float dotProduct = 0;
+            switch (ccol)
+            {
+                case 3:
+                    dotProduct += pmat[i * ccol + 2] * psrc[2];
```

Comment:
(minor) It may be beneficial here to note that you are falling through the case statements. I'm not sure...

Reply:
yes that is intentional, I can add a comment for it. We are doing a similar thing in scale and add operations